projects/ad-acevsecrdset-sl/src/inter/inter.c

```diff
@@ -0,0 +1,375 @@
+	if (ret) {
+		printf("ERROR %d \n", ret);
+	}
+}
+
+/**
+ * @brief Get GPIO flag value
+ *
+ * @return flag value
+ */
+int get_gpio_flag_state(void)
+{
+	return gpio_flag;
+}
+
+/**
+ * @brief Reset GPIO low flag value
+ *
+ * @return none
+ */
+void reset_gpio_low_flag_state(void)
+{
+	gpio_flag = 0;
+}
+
+/**
+ * @brief Get GPIO Opto_out1 flag value
+ *
+ * @return flag value
+ */
+int get_gpio_opto_out1_flag_state(void)
+{
+	return opto_out_1_flag;
+}
+
+/**
+ * @brief Reset GPIO Opto_out1 flag value
+ *
+ * @return none
+ */
+void reset_gpio_opto_out1_flag_state(void)
+{
+	opto_out_1_flag = 0;
+}
+
+/**
+ * @brief Get GPIO Opto_out2 flag value
+ *
+ * @return flag value
+ */
+int get_gpio_opto_out2_flag_state(void)
+{
+	return opto_out_2_flag;
+}
+
+/**
+ * @brief Reset GPIO Opto_out2 low flag value
+ *
+ * @return none
+ */
+void reset_gpio_opto_out2_flag_state(void)
+{
+	opto_out_2_flag = 0;
+}
+
+/**
+ * @brief Initialize inter phase
+ *
+ * @return 0 in case of success, error code otherwise
+ */
+int inter_init(struct stout *stout)
+{
+	struct no_os_irq_ctrl_desc *irq_desc1;
+	struct no_os_irq_ctrl_desc *irq_desc2;
+	struct no_os_irq_ctrl_desc *irq_desc3;
+	struct no_os_gpio_desc *pin;
+	struct no_os_gpio_desc *pin2;
+	struct no_os_gpio_desc *pin3;
+	int ret;
+
+	/* Setup GPIO interrupts */
+	struct no_os_callback_desc p2_cb = {
+		/** Callback to be called when the event occurs. */
+		.callback = gpio_2_3_cb_fn,
+		/** Parameter to be passed when the callback is called */
+		.ctx = NULL,
+		/** Event that triggers the calling of the callback. */
+		.event = NO_OS_EVT_GPIO,
+		/** Interrupt source peripheral specifier. */
+		.peripheral = NO_OS_GPIO_IRQ,
+		/** Not used in the case of a GPIO IRQ controller */
+		.handle = NULL
+	};
+
+	/* Setup GPIO interrupts */
+	struct no_os_callback_desc p2_opt1_cb = {
+		/** Callback to be called when the event occurs. */
+		.callback = gpio_2_5_cb_fn,
+		/** Parameter to be passed when the callback is called */
+		.ctx = NULL,
+		/** Event that triggers the calling of the callback. */
+		.event = NO_OS_EVT_GPIO,
+		/** Interrupt source peripheral specifier. */
+		.peripheral = NO_OS_GPIO_IRQ,
+		/** Not used in the case of a GPIO IRQ controller */
+		.handle = NULL
+	};
+
+	/* Setup GPIO interrupts */
+	struct no_os_callback_desc p2_opt2_cb = {
+		/** Callback to be called when the event occurs. */
+		.callback = gpio_2_7_cb_fn,
+		/** Parameter to be passed when the callback is called */
+		.ctx = NULL,
+		/** Event that triggers the calling of the callback. */
+		.event = NO_OS_EVT_GPIO,
+		/** Interrupt source peripheral specifier. */
+		.peripheral = NO_OS_GPIO_IRQ,
+		/** Not used in the case of a GPIO IRQ controller */
+		.handle = NULL
+	};
+
+	/* Port 2 pin */
+	ret = no_os_gpio_get(&pin, &gpio_ip);
+	if (ret) {
+		printf("ERROR\n");
+		return ret;
+	}
+
+	ret = no_os_gpio_direction_input(pin);
+	if (ret) {
+		printf("ERROR\n");
+		return ret;
```

Comment:
where is the memory deallocated for the `pin` in case of error? same applies for the rest of the gpio/irq descriptors.